### PR TITLE
Case insensitive checkout improvements

### PR DESCRIPTION
**Pull request by ethomson**
On Windows, you might sloppily rewrite a file (or have a sloppy text editor that does it for you) and accidentally change its case.  (eg, "README" -> "readme").  Git ignores this accidental case changing rename during checkout and will happily write the new content to the file despite the name change.  We should, too.

Further, ensure that on a case insensitive filesystem that we can checkout into some folder 'FOLDER' that exists on disk, even if the target of the checkout is a different case (eg 'folder').

Revert prior change to checkout with the case sensitive iterator.  Now, when checking out with a case-insensitive working directory, we want to change the case of items in the working directory to reflect changes that occured in the checkout target.  Diff now has an option to break case-changing renames into delete/add to enable this.

### CODE REVIEW

File: src/diff.c [hunk 1]
```diff
@@ -822,6 +822,19 @@ static int maybe_modified(
...
+	 * and an add of the new so that consumers can act accordingly (eg,
+	 * checkout will update the case on disk.)
+	 */
+	if (DIFF_FLAG_IS_SET(diff, GIT_DIFF_IGNORE_CASE) &&
+		DIFF_FLAG_IS_SET(diff, GIT_DIFF_INCLUDE_CASECHANGE) &&
+		strcmp(oitem->path, nitem->path) != 0) {
+
+		if (!(error = diff_delta__from_one(diff, GIT_DELTA_DELETED, oitem)))
+			error = diff_delta__from_one(diff, GIT_DELTA_ADDED, nitem);
+			return error;
```
Comment (whoisj) [inline src/diff.c]: Are you missing braces here or is the indentation just odd in the HTML review?

Comment (ethomson) [inline src/diff.c]: No; my indentation was actually wrong.  Thanks.  Fixed.

**Comment by carlosmn** [comment]
What's the advantage to changing the `test_realpath()` implementation? AFAICT this is not what we do on Mac or anywhere else for `p_realpath()`. It would be great for that commit to explain what's wrong with the way we do it now.

**Comment by ethomson** [comment]
Yep, that was sort of weak.  I added some details to the final commit's message.  Thanks.

**Pull request merged**